components/shared/Search.tsx

```diff
@@ -1,33 +1,158 @@
+    }
+  };
   return (
-    <div className="bg-black-700 flex items-center gap-2 rounded-md px-2">
-      <Image
-        src={searchIcon}
-        alt="Search Icon"
-        className="pointer-events-none"
-      />
-      <input
-        className="placeholder:paragraph-4-medium placeholder:text-white-500 bg-black-700 w-full rounded-md border-none text-sm focus:outline-none"
-        placeholder="Search..."
-        onChange={(e) => {
-          handleSearch(e.target.value);
-        }}
-      />
-      <Image
-        src={shortcutIcon}
-        alt="Shortcut Icon"
-        className="pointer-events-none"
-      />
-    </div>
+    <>
+      <div
+        className="bg-black-700 paragraph-4-medium flex cursor-pointer items-center justify-between rounded-md p-4"
+        onClick={() => setOpen((open) => !open)}
+      >
+        <div className="flex gap-x-2">
+          <Image
+            src={searchIcon}
+            alt="Search Icon"
+            className="pointer-events-none"
+          />
+          <p className="text-white-500">Search...</p>
+        </div>
+        <Image
+          src={shortcutIcon}
+          alt="Shortcut Icon"
+          className="pointer-events-none"
+        />
+      </div>
+
+      <Command.Dialog
+        open={open}
+        onOpenChange={setOpen}
+        label="Global Command Menu"
+        className="fixed inset-0 z-50 flex h-full w-full items-center justify-center backdrop-blur"
+      >
+        <div className="bg-black-800 flex w-1/2 flex-col">
+          <div className="bg-black-700 flex w-full items-center gap-x-2 border-none p-4  py-3">
+            <Image
+              src={searchIcon}
+              alt="Search Icon"
+              className=" pointer-events-none ml-2"
+            />
+            <Command.Input
+              value={searchTerm}
+              onValueChange={setSearchTerm}
+              className="bg-black-700 paragraph-3-regular text-white-300 placeholder:text-white-300 w-full border-none p-0 py-1"
+              placeholder="Type a command or search..."
+            />
+            <div className="bg-black-800 text-white-300 paragraph-4-regular rounded p-1">
+              ESC
+            </div>
+          </div>
+          <Command.List className="paragraph-3-regular text-white-300 h-64 overflow-auto p-4">
+            <Command.Empty>No results found.</Command.Empty>
+            <Command.Group>
+              {posts &&
+                posts.length > 0 &&
+                posts.map((post) => {
+                  return (
+                    <Link
+                      key={post.id}
+                      href={`/posts/${post.id}`}
+                      onClick={() => setOpen((open) => !open)}
+                    >
+                      <Command.Item className="hover:bg-black-600 flex cursor-pointer items-center gap-x-2 px-2 py-2 hover:rounded hover:py-2">
+                        {iconMatch(post)}
+                      </Command.Item>
+                    </Link>
```

Comment:
Is this the best way to link to a post detail when a search item is clicked on?

Reply:
Yep. We're linking to the page we want to go to and closing the modal.

Was there something in particular you didn't like about this?

Reply:
I just wasn't sure if cmd+k provided something in their library that I had missed for this